client/src/hooks/datafiles/mutations/useCompress.ts

```diff
@@ -0,0 +1,209 @@
+};
+
+export type TConfigurationValues = {
+  execSystemId?: string;
+  execSystemLogicalQueue?: string;
+  maxMinutes?: number;
+  nodeCount?: number;
+  coresPerNode?: number;
+  allocation?: string;
+  memoryMB?: number;
+};
+
+export type TOutputValues = {
+  name: string;
+  archiveSystemId?: string;
+  archiveSystemDir?: string;
+};
+
+export interface TJobSubmit extends TConfigurationValues, TOutputValues {
+  archiveOnAppError?: boolean;
+  appId: string;
+  fileInputs?: TAppFileInput[];
+  parameterSet?: TParameterSetSubmit;
+}
+
+export type TJobBody = {
+  operation?: TJobPostOperations;
+  uuid?: string;
+  job: TJobSubmit;
+  licenseType?: string;
+  isInteractive?: boolean;
+  execSystemId?: string;
+};
+
+interface IJobPostResponse extends TTapisJob {
+  execSys?: TTapisSystem;
+}
+
+type TJobPostResponse = {
+  response: IJobPostResponse;
+  status: number;
+};
+
+async function submitJobUtil(body: TJobBody) {
+  const res = await apiClient.post<TJobPostResponse>(
+    `/api/workspace/jobs`,
+    body
+  );
+  return res.data.response;
+}
+
+function useCompress() {
+  const dispatch = useDispatch();
+  const status = useSelector(
+    (state: any) => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const setStatus = (newStatus: any) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: newStatus, operation: 'compress' },
+    });
+  };
+
+  const compressErrorAction = (errorMessage: any) => {
+    return {
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: {
+        status: { type: 'ERROR', message: errorMessage },
+        operation: 'compress',
+      },
+    };
+  };
+
+  const compressApp = useSelector(
+    (state: any) => state.workbench.config.compressApp
+  );
+
+  const defaultAllocation = useSelector(
+    (state: any) =>
+      state.allocations.portal_alloc || state.allocations.active[0].projectName
+  );
+
+  const systems = useSelector(
+    (state: any) => state.systems.storage.configuration
+  );
+
+  const { mutateAsync } = useMutation({ mutationFn: submitJobUtil });
+
+  const compress = ({
+    scheme,
+    files,
+    filename,
+    compressionType,
+  }: {
+    scheme: string;
+    files: TTapisFile[];
+    filename: string;
+    compressionType: string;
+  }) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'RUNNING', operation: 'compress' },
+    });
+
+    let defaultPrivateSystem: TPortalSystem | undefined;
+
+    if (scheme !== 'private' && scheme !== 'projects') {
+      defaultPrivateSystem = systems.find((s: any) => s.default);
+
+      if (!defaultPrivateSystem) {
+        throw new Error('Folder downloads are unavailable in this portal', {
+          cause: 'compressError',
+        });
+      }
+    }
+
+    const params = getCompressParams(
+      files,
+      filename,
+      compressionType,
+      compressApp,
+      defaultAllocation,
+      defaultPrivateSystem
```

Comment:
drop `defaultPrivateSystem` to use path and system of the `files`

Reply:
Actually, we want to use `defaultPrivateSystem` unless the origin scheme is `private`, and origin `api` is `tapis`

Reply:
`defaultPrivateSystem: files[0].scheme === 'private' && files[0].api === 'tapis' ? null : defaultPrivateSystem

Reply:
I have this logic in there now.